morpho-checkpoint-node/src/Morpho/Tracing/Pretty.hs

```diff
@@ -1,81 +1,132 @@
 import qualified Ouroboros.Network.NodeToClient as NtC
 import Ouroboros.Network.NodeToNode
-  ( DnsTrace (..),
+  ( AcceptConnectionsPolicyTrace (ServerTraceAcceptConnectionHardLimit, ServerTraceAcceptConnectionRateLimiting),
+    DnsTrace (..),
     ErrorPolicyTrace (..),
+    LocalAddresses (LocalAddresses),
     SubscriptionTrace (..),
     WithAddr (..),
     WithDomainName (..),
     WithIPList (..),
   )
 import qualified Ouroboros.Network.NodeToNode as NtN
-import Ouroboros.Network.PeerSelection.LedgerPeers (TraceLedgerPeers)
-import Ouroboros.Network.TxSubmission.Inbound (TraceTxSubmissionInbound)
-import Ouroboros.Network.TxSubmission.Outbound (TraceTxSubmissionOutbound)
-import Prettyprinter (Doc, Pretty (..), list, viaShow, (<+>))
+import Ouroboros.Network.PeerSelection.LedgerPeers (AccPoolStake (AccPoolStake), IP (IPv4, IPv6), PoolStake (PoolStake), RelayAddress (RelayAddressAddr, RelayAddressDomain), TraceLedgerPeers (FetchingNewLedgerState, PickedPeer, PickedPeers))
+import Ouroboros.Network.PeerSelection.RootPeersDNS (DomainAddress (DomainAddress))
+import Ouroboros.Network.Protocol.BlockFetch.Type (ClientHasAgency, Message, ServerHasAgency)
+import Ouroboros.Network.Subscription (ConnectResult (ConnectSuccess, ConnectSuccessLast, ConnectValencyExceeded))
+import Ouroboros.Network.TxSubmission.Inbound (ProcessedTxCount (ProcessedTxCount), TraceTxSubmissionInbound (TraceTxInboundCanRequestMoreTxs, TraceTxInboundCannotRequestMoreTxs, TraceTxInboundTerminated, TraceTxSubmissionCollected, TraceTxSubmissionProcessed))
+import Ouroboros.Network.TxSubmission.Outbound (TraceTxSubmissionOutbound (TraceControlMessage, TraceTxSubmissionOutboundRecvMsgRequestTxs, TraceTxSubmissionOutboundSendMsgReplyTxs))
+import Prettyprinter (Doc, Pretty (..), comma, concatWith, hsep, punctuate, viaShow, (<+>))
+import qualified Prelude
 
 class MPretty a where
   mpretty :: a -> Doc ann
 
-instance (HashAlgorithm h, BftCrypto c) => MPretty (MorphoBlock h c) where
-  mpretty MorphoBlock {..} =
-    "("
-      <> mpretty morphoPrev
-      <> "->"
-      <> mpretty morphoHeaderHash
-      <> ","
-      <> mpretty morphoSlotNo
-      <> ","
-      <> mpretty morphoTxs
+instance MPretty (HeaderHash blk) => MPretty (MorphoState blk) where
+  mpretty (MorphoState lastCkpt ckptAt votes tip) =
+    "MorphoState(morphoTip:"
+      <+> mpretty tip
+      <> ", lastCheckpoint:"
+      <+> mpretty lastCkpt
+      <> ", checkpointAt:"
+      <+> mpretty ckptAt
+      <> ", currentVotes:"
+      <+> mpretty votes
       <> ")"
-    where
-      MorphoHeader {..} = morphoHeader
-      MorphoStdHeader {..} = morphoHeaderStd
-      MorphoBody {..} = morphoBody
 
-instance
-  (HashAlgorithm h, BftCrypto c) =>
-  MPretty (Header (MorphoBlock h c))
-  where
+instance MPretty PublicKey where
+  mpretty (PublicKey bytes) =
+    "PublicKey(" <> viaShow (take 8 $ Prelude.show bytes) <> ")"
```

Comment:
This currently prints `"` around the bytes.

I suggest:
- Creating a `Morpho.Common.Bytes.bytesToHex :: Bytes -> Text`, which is based on the `Show Bytes` implementation
- Using `bytesToHex` for `Show Bytes`
- Using `bytesToHex` to implement `MPretty Bytes`
- Then using `mpretty bytes` in the implementation of `MPretty PublicKey`

Reply:
Just found a `bytesToHex` in Morpho.Common.Conversions`. However, I can't use that function in `Show Bytes` since it introduces a cyclic dependency. But I can use it for tracing.